lib/src/settings.rs

```diff
@@ -98,19 +98,14 @@ impl UserSettings {
     }
 
     pub fn user_name(&self) -> String {
-        self.config
-            .get_string("user.name")
-            .unwrap_or_else(|_| Self::user_name_placeholder().to_string())
+        self.config.get_string("user.name").unwrap_or_default()
     }
 
-    pub fn user_name_placeholder() -> &'static str {
-        "(no name configured)"
-    }
+    // Must not be changed to avoid git pushing older commits with no set name
```

Comment:
We typically drop support for old formats after 6 months or so, but most upgrades so far have automatically upgraded existing repo. In this case, I don't think we'll want to rewrite existing commits, so I guess we should keep support for these old commits much longer. It's not very costly in terms of maintenance anyway. I'm mostly thinking out loud, but let me know if you disagree.

Reply:
I agree we should continue supporting the old placeholders. Rewriting commits(' hashes) would be very distributive.